Common/Orders/Fills/IFillModelInterface.cs

```diff
@@ -0,0 +1,32 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+
+namespace QuantConnect.Orders.Fills
+{
+    /// <summary>
+    /// Represents an interface that simulates order fill events
+    /// </summary>
+    /// <remarks>Users should override base class <see cref="FillModel"/></remarks>
+    public interface IFillModelInterface
+    {
+        /// <summary>
+        /// Return an order event with the fill details
+        /// </summary>
+        /// <param name="context">A context object containing the security and order</param>
+        /// <returns>Order fill information detailing the average price and quantity filled.</returns>
+        OrderEvent Fill(FillModelContext context);
```

Comment:
Might make sense to give `IFillModelInterface.Fill` a dedicated response object as was done w/ various other methods converted to the context pattern. The idea being that the dedicated return type allows it to evolve. Also, the dedicated return type is little more than a wrapper object to maintain compile-time compatibility. This could make it easier to support multiple fills at a single time step or other (unknown) additions to the return in the future.